image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -146,161 +138,133 @@ def rhel_subscription(username, password, pool, image, proxy_hostname=None, prox
     ])
-    virt_customize_run(
-        'nuage_packages -a %s --memsize %s --selinux-relabel' % (
-        image, VIRT_CUSTOMIZE_MEMSIZE))
+    virt_customize_run('nuage_packages -a %s --memsize %s --selinux-relabel' %
+                       (image, VIRT_CUSTOMIZE_MEMSIZE))
     cmds_run(['rm -f nuage_packages'])
 
-#####
-# Function to create the repo file
-#####
-
-def create_repo_file(reponame, repoUrl, image, gpgcheck):
-    create_repo = \
-        'cat <<EOT > create_repo \n' \
-        'touch /etc/yum.repos.d/nuage.repo \n' \
-        'echo "[Nuage]" >> /etc/yum.repos.d/nuage.repo \n' \
-        'echo "name=%s" >> /etc/yum.repos.d/nuage.repo \n' \
-        'echo "baseurl=%s" >> /etc/yum.repos.d/nuage.repo \n' \
-        'echo "enabled = 1" >> /etc/yum.repos.d/nuage.repo \n' \
-        'echo "gpgcheck = %s" >> /etc/yum.repos.d/nuage.repo \n' \
-        'EOT' % (reponame, repoUrl, gpgcheck)
-
-    cmds_run([create_repo])
-    virt_customize_run(
-        'create_repo -a %s --memsize %s --selinux-relabel' % (
-        image, VIRT_CUSTOMIZE_MEMSIZE))
-    cmds_run(['rm -f create_repo'])
-
-
-#####
-# Function to clean up the repo file
-#####
-def delete_repo_file(image):
-    virt_customize(
-        '"rm -f /etc/yum.repos.d/nuage.repo" -a %s --selinux-relabel'
-        % (image))
-
 
 #####
 # Function to install Nuage AVRS packages that are required
 #####
 
-def copy_avrs_packages(image, proxy_hostname=None, proxy_port=None):
-    if proxy_hostname != None and proxy_port != None:
-        avrs_cmds = 'cat <<EOT > nuage_avrs_packages \n' \
-                    'export http_proxy=http://%s:%s \n' \
-                    'export https_proxy=http://%s:%s \n' % (
-                    proxy_hostname, proxy_port, proxy_hostname,
-                    proxy_port)
-    else:
-        avrs_cmds = 'cat <<EOT > nuage_avrs_packages \n'
-
-    install_cmds = \
-        'mkdir ./6wind \n ' \
-        'rm -rf /var/cache/yum/Nuage \n' \
-        'yum clean all \n' \
-        'touch /kernel-version \n' \
-        'rpm -q kernel | awk \'{ print substr(\$1,8) }\' ' \
-        '> /kernel-version \n' \
-        'yum install -y createrepo \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        'kernel-headers-\$(cat /kernel-version) ' \
-        'kernel-devel-\$(cat /kernel-version) ' \
-        'kernel-debug-devel-\$(cat /kernel-version) ' \
-        'python-pyelftools* dkms* 6windgate* ' \
-        'nuage-openvswitch nuage-metadata-agent virtual-accelerator* \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        'selinux-policy-nuage-avrs* \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        '6wind-openstack-extensions \n' \
-        'rm -rf /kernel-version \n' \
-        'EOT'
-    avrs_cmds = avrs_cmds + install_cmds
-
-    cmds_run([avrs_cmds])
+
+def copy_avrs_packages(image, nuage_avrs_repos, proxy_hostname=None, proxy_port=None):
+    enable_repos_cmd = "subscription-manager repos"
```

Comment:
`subscription-manager repos` can be only used for redhat repos but not custom repos.
For custom repos, we need to use `yum-config-manager`
```
[root@rhdirector avrs]# subscription-manager repos --enable=nuage_openstack
1 local certificate has been deleted.
Error: 'nuage_openstack' does not match a valid repository ID. Use "subscription-manager repos --list" to see valid repositories.
```

when I ran `yum-config-manager --enable nuage_openstack`, this repo got enabled